pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -536,6 +536,19 @@ func (r *Replica) applySnapshot(
 		}
 	}
 
+	for _, sr := range subsumedRepls {
```

Comment:
Consider moving it slightly more down, before the "clear subsumed in-memory" part, so that `applySnapshot` has clearer phases: writing state, loading state, in-memory stuff.

`clearSubsumedReplicaDiskData` only uses `Desc` of the ranges, so these changes can seemingly be done in any order.

Reply:
There is a dependency here, we must mark the subsumed replicas as destroyed before ingesting the SSTs. So there'll necessarily be some interleaving between the storage interactions and the in-memory updates, where the in-memory updates occur before and after the I/O.
It might make sense to mark as destroyed right before ingesting, but I'll leave as is for now.

Reply:
Added a comment though.